lib/sidebar/index.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { IFolder, INode } from '../node/index.ts'
+import logger from '../utils/logger.ts'
+
+interface SidebarUpdateContext {
+	/**
+	 * The currently selected node for which the sidebar is shown.
+	 */
+	node: INode
+
+	/**
+	 * The parent of the current selected node.
+	 * This is not necessarily the real parent folder of the node in means of the real filesystem tree,
+	 * but rather the parent folder in the current view of the files app.
+	 */
+	parent: IFolder
+}
+
+/**
+ * Implementation of a custom sidebar tab within the files app.
+ */
+export interface ISidebarTab {
+	/**
+	 * Unique id of the sidebar tab.
+	 * This has to conform to the HTML id attribute specification.
+	 */
+	id: string
+
+	/**
+	 * The localized name of the sidebar tab.
+	 */
+	displayName: string
+
+	/**
+	 * The icon, as SVG, of the sidebar tab.
+	 */
+	iconSvg: string
+
+	/**
+	 * The order of this tab.
+	 * Use a low number to make this tab ordered in front.
+	 */
+	order: number
+
+	/**
+	 * Callback to check if the sidebar tab should be shown for the selected node.
+	 *
+	 * @param node - The currently selected node
+	 */
+	enabled: (node: INode) => boolean
+
+	/**
+	 * Called by the files app if this tab has become the active tab or was deactivated.
+	 *
+	 * @param active - The new active state of this tab
+	 */
+	setActive: (active: boolean) => void | Promise<void>
```

Comment:
So, we've been using events lately since you introduced them :)
Maybe it's time to make it a common pattern at Nextcloud?

Reply:
Ah, yeah :facepalm: